odxtools/parentref.py

```diff
@@ -15,101 +14,106 @@
+        self_prio = PRIORITY_OF_DIAG_LAYER_TYPE[self.self_layer_type]
+        parent_prio = PRIORITY_OF_DIAG_LAYER_TYPE[self.parent_layer_type]
+        assert self_prio > parent_prio, "diagnostic layers can only inherit from layers of lower priority"
+
+    @property
+    def parent_layer(self) -> "DiagLayer":
+        return self._parent_layer
 
     @staticmethod
-    def from_et(et_element, doc_frags: List[OdxDocFragment]) -> "ParentRef":
+    def from_et(et_element, self_layer_type: DiagLayerType,
+                doc_frags: List[OdxDocFragment]) -> "ParentRef":
 
-        parent_ref = OdxLinkRef.from_et(et_element, doc_frags)
-        assert parent_ref is not None
+        parent_layer_ref = OdxLinkRef.from_et(et_element, doc_frags)
+        assert parent_layer_ref is not None
 
         not_inherited_diag_comms = [
-            el.get("SHORT-NAME") for el in et_element.iterfind(
-                "NOT-INHERITED-DIAG-COMMS/NOT-INHERITED-DIAG-COMM/DIAG-COMM-SNREF")
+            el.get("SHORT-NAME") for el in et_element.iterfind("NOT-INHERITED-DIAG-COMMS/"
+                                                               "NOT-INHERITED-DIAG-COMM/"
+                                                               "DIAG-COMM-SNREF")
         ]
         not_inherited_dops = [
-            el.get("SHORT-NAME")
-            for el in et_element.iterfind("NOT-INHERITED-DOPS/NOT-INHERITED-DOP/DOP-BASE-SNREF")
+            el.get("SHORT-NAME") for el in et_element.iterfind("NOT-INHERITED-DOPS/"
+                                                               "NOT-INHERITED-DOP/"
+                                                               "DOP-BASE-SNREF")
+        ]
+        not_inherited_tables = [
+            el.get("SHORT-NAME") for el in et_element.iterfind("NOT-INHERITED-TABLES/"
+                                                               "NOT-INHERITED-TABLE/"
+                                                               "TABLE-SNREF")
+        ]
+        not_inherited_variables = [
+            el.get("SHORT-NAME") for el in et_element.iterfind("NOT-INHERITED-VARIABLES/"
+                                                               "NOT-INHERITED-VARIABLE/"
+                                                               "DIAG-VARIABLE-SNREF")
+        ]
+
+        not_inherited_global_neg_responses = [
+            el.get("SHORT-NAME") for el in et_element.iterfind("NOT-INHERITED-GLOBAL-NEG-RESPONSES/"
+                                                               "NOT-INHERITED-GLOBAL-NEG-RESPONSE/"
+                                                               "GLOBAL-NEG-RESPONSE-SNREF")
         ]
-        ref_type = et_element.get(f"{xsi}type")
-        ref_type = ref_type[:-len("-REF")]
+
+        # determine the type of the referenced diag layer. for
+        # this, we need to strip the '-REF' suffix of the
+        # element's {xsi}type attribute
+        parent_layer_type_str = et_element.get(f"{xsi}type")
```

Comment:
Sure all have `xsi:type` attribute?

Reply:
if I interpret the XSD correctly, these tags could be plain PARENT-REF, but then the XSD types would be useless. (I guess that's a minor bug in the spec.) anyway, I changed this to make it work unconditionally: 1a204e38faf90f3082b3eb5fb4354797681a5fbf